### DISCUSSION TITLE
UI in general and where to now?

### DISCUSSION THREAD

**Discussion by bofdahof** [FreeCAD/FreeCAD#17478] [Discussions that will turn in to issues but need refining]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

I have now provided a bunch of somewhat detailed UI issue reports.

So far it is like shooting ducks in a pond. Pick up any dialog and try to understand it, and then try to understand what makes it feel difficult or illogical or unusual or just dumb.

From what I see there is unlikely to be a meaningful response, and what response there is will likely be an opinion fest on specific detail. This is wasteful. 

I believe there is some sort of chat group who discuss these things, but I see no concrete results or input. Surely their mandate should be to assertively provide expertise to guide development and standards and guidelines for devs to follow. Devs are notoriously bad at UI.

There are those that say CAD is hard. But if the UI is bad, and it is, then it makes CAD really really hard. And UI is definitely not easy because it deals first hand with people.

Sorry to sound negative, but I have put a lot of effort into this. It is depressing to see the same lack of empathy for users repeated everywhere, right across the board. Chasing individual flaws is like whack-a-mole. So I won't do any more.

IMHO a much more in-depth dialog is needed if the goal is for users to achieve as pleasant an experience as the marketing promises.

### Full version info

```shell
N/A
```

### Subproject(s) affected?

None

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by luzpaz** [comment]
Thank you in a very big way for your UI/UX audit. I and many others I'm sure are appreciative for the in-depth work you're doing. Here are some key points I've observed over the years of being involved with FLOSS projects:

* UI is very frustrating to work with because one is at the mercy of the software framework being used, in our case it's Qt. Which is still ironing out it's issues. Programmers can spend days/weeks wrestling with minutia that quickly leads to burn out. Especially for unpaid volunteers.
* Because of this, very often there is an approach to get an MVP proposal out to get the ball rolling with the intention of improving the subproject on an iteration by iteration basis later.  And as usual, later doesn't always come for the person who spearheaded the original push as life intervenes and can call them elsewhere. Usually another volunteer will come to 'pick up the torch' but they have different strengths/skills and approaches.
* UI/UX is very subjective. Discussions can quickly escalate in to ad hominem attacks (directed against a person rather than the position they are maintaining). I've noticed (speaking very broadly now) that designers have a tendency to have very strong opinions and not play well with others. This is problematic in an opensource volunteer based project. group dynamics and cohesion are key. They supersede technical know-how, impressive resumes, esthetics. and raw passion. People don't get into FLOSS for more drama, they have enough of that in other parts of their life. Unfortunately, not all humans know how to play well with others. 
* UI/UX is very often subjective and so it takes quite a long process to find a common decision amongst everyone involved in the decision making. This process of debate can be very time-consuming and for folks that are more on the impatient problem solving side of the equation - they don't last long. There is a perseverance that is required.
* I read this somewhere that designers don't stick around long in open source because their expertise is quite expensive and they can't justify spending so much time, as mentioned in previous points, in a volunteer status. So they end up moving on. FLOSS is full of examples of this.  
* Often outsiders looking in think that the state of the UI is how the programmers think of others. If the UI is clunky or sloppy, they will take it personally "Wow, they must think very little of us to treat us this way". It's as if they are interpreting a narrative that is demeaning to them because of unsophisticated and amateurish UI/UX. This is a very subjective perspective and uninformed approach. But for someone who is attracted to esthetics and feel 'assaulted' by creative decisions may be too busy experiencing the distress they feel internally from the experience then fully comprehending that not everyone responds that way and that the intention could still be a work in progress. 

Finally, this is why the FreeCAD community has a Design Advisory Group and they hang out on discord. There is thinking going on around the UI/UX and it's more organized than it's ever been.

**Comment by bofdahof** [comment]
@luzpaz Thank you for long comment.

I'm not sure the cost of designers is really relevant. Good engineers and good designers are all expensive. 

Of the many issues I documented, I don't think any could be laid at the feet of the framework. They are pretty much all design and implementation details.

I'm not sure that having a go at outsiders looking in is at all helpful. People are perfectly entitled to their feelings, and equally entitled to express them. Judging or presuming their attitude seems unnecessarily defensive, and unreasonable. Open source is not a protected species. Like any product, embracing customers will yield better results than mocking them.

Casting aspersions about designers is also not helpful. Most artists are famously odd-bods and might be expected to react badly to rigid control structures. But they are vital. So unless the control structure adjusts ...

I agree re the divide between programmers and designers. It is legion. Different parts of the brain or something. The  programmer targets the processor. Quite black and white. Even if behind the scenes the code is ugly. The processor doesn't care. OTOH The designer targets a pleasurable user experience (UX) for humans, via a user interface (UI). Both require care and skill, albeit differently.

To say UX/UI is subjective is valid in the sense that human expectations are different, so the only valid metric is user opinion. In the case of FreeCAD the feedback is overwhelmingly that the UX is sub-par. 

Why though? Users themselves are unlikely to know. They just get a feeling. The trick, if there is one, is to understand why. This starts by being able to identify an issue, and then identifying the cause of the issue and finally developing techniques to avoid the issue in the first place. Many programmers seem to stumble at the first stage. Many more at the second. Few programmers have meaningful training or experience with design. And many designers will have limited programming skills.

It is worth noting that in many environments it is the norm that the design is done before programming begins, often via mock-ups etc. Code is only relevant to the extent of limitations it may impose.

In FreeCAD code is written and then various programmers come along and patch it up. Quite messy.

The committee problem: UI discussions are easily dominated by strong opinions of those lacking that deep or intuitive understanding that is the mark of the good designer. Likely the designer will quickly depart. If the members do sufficiently respect the designer then the committee is superfluous anyway.
 
At the end of all that the challenges are clear. How to bridge the void between programmer and designer, and how to encourage more designers to play?

**Reply by luzpaz** [reply]
![Screenshot_20241030_143900](https://github.com/user-attachments/assets/d4f32b0a-3154-45fe-b375-97ecc6df056f)
Source: https://x.com/BillieRubenMake/status/1481028985585016835

**Comment by bofdahof** [comment]
The question remains: What to do about it? Can't just make excuses and accept poor UI. For just about any argument one can find something on the web to support it, but in the mean time ...

**Reply by luzpaz** [reply]
I meant to respond in more depth but had little time. Essentially, I think we're in agreement. 

I also think this discussion makes more sense in the context FreeCAD Design Advisory Group. Have you contacted them yet on Discord ?

**Reply by bofdahof** [reply]
In fairness, re-reading this your quotes were almost entirely in agreement with what I stated previously, so yeah, essentially you are in agreement. Which is cool. The challenge lies in the last part of the last quote re protective of being paid. I say perhaps, but aren't we all?

**Reply by bofdahof** [reply]
> FreeCAD Design Advisory Group. Have you contacted them yet on Discord

I have seen little to enamour me to Discord. Maybe because I am old (some say not old, just experienced!) what I see there is a stream of discussion about anything and everything rather than a topic-based structure seeking clear outcomes. Feels more like a social chat group, probably used primarily via smartphone.

**Comment by bofdahof** [comment]
> Essentially, I think we're in agreement.

Great!

> FreeCAD Design Advisory Group

See previous remarks. In the past there were those with opinions. Now there are those with opinions and a title.
